pkg/types/azure/validation/disk.go

```diff
@@ -0,0 +1,45 @@
+package validation
+
+import (
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types/azure"
+)
+
+// ValidateDiskEncryption checks that the specified disk encryption configuration is valid.
+func ValidateDiskEncryption(p *azure.MachinePool, cloudName azure.CloudEnvironment, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	childFldPath := fldPath.Child("osDisk", "diskEncryptionSet")
+
+	diskEncryptionSet := p.OSDisk.DiskEncryptionSet
+	diskEncryptionSetID := diskEncryptionSet.ToID()
```

Comment:
At this stage, we can't assume subscription is set and therefore have a valid id.